extensions/script-libraries/qr/qr.livecodescript

```diff
@@ -0,0 +1,2443 @@
+      local tRefAlpha
+      put _alpha(tErrorCodeA[tRefExp]) into tRefAlpha
+      repeat with i = 1 to tGenExpCount
+         local tECIndex, tGenAlpha
+         put tRefExp - i + 1 into tECIndex
+         put item i of tGenExp into tGenAlpha
+         # calculate the new coefficient value
+         put _int((tRefAlpha + tGenAlpha) mod 255) bitXOr tErrorCodeA[tECIndex] into tErrorCodeA[tECIndex]
+      end repeat
+      # remove leading zero coefficients
+      repeat with i = tRefExp to 0 step -1
+         if tErrorCodeA[i] = 0 then
+            delete variable tErrorCodeA[i]
+            subtract 1 from tRefExp
+         else
+            exit repeat
+         end if
+      end repeat
+   end repeat
+   
+   # return the error codes as a binary stream
+   local tKeys
+   put the keys of tErrorCodeA into tKeys
+   sort lines of tKeys descending numeric
+   local tErrorBytes
+   repeat for each line tKey in tKeys
+      put numToChar(tErrorCodeA[tKey]) after tErrorBytes
+   end repeat
+   return tErrorBytes
+end _errorCodes
+
+/**
+
+Create a QR code
+
+Example:
+qrCreate "MyImage", "QR data", "M", 3
+
+Description:
+Set the text of an image to a QR code or get the PMG data of the QR code image.
+
+Parameters:
+pImg:
+The name of an image object on the card, a filename or the special code "!" which
+will return the image data in the result.
+
+pData:
+The data to be encoded.
+
+pECC (enum):
+The error correction level
+
+- "L" : 7% error correction
+- "M" : 15% error correction
+- "Q" : 25% error correction
+- "H" : 30% error correction
+
+pSize (integer):
+Module size you can make the code larger/smaller.  You can also just
+rescale the image after the code has been created.
+
+pMask (integer):
+An integer value between 0 and 7. When encoding a QR code, there are
+eight mask patterns that you can use to change the outputted matrix.
+Each mask pattern changes the bits according to their coordinates in
+the QR matrix. The purpose of a mask pattern is to make the QR code
+easier for a QR scanner to read.
+
+The result:
+If successful, the first line of the result will be the following values: 
+- QR version
+- ECC
+- Mode
+- Mask
+- Image size with quiet zone (based on module size 1)
+
+If there was an error the result will contain "Error:" and then an error
+description.
+
+If the special character '!' is used for pImg, then the result also
+contains the raw output of a png image.  You can use this data, for example,
+with LiveCode server to generate QR codes on web pages.
+
+*/
+
+command qrCreate pImg, pData, pECC, pSize, pMask
+   lock screen
+   
+   try
+      
+      # check target image object exists
+      local tPNG
+      if there is an img pImg then
+         put the long id of img pImg into pImg
+         put false into tPNG
+      else
+         put not exists(pImg) into tPNG
+      end if
+      
+      # check ECC
+      if not(pECC is among the items of "L,M,Q,H") then
+         return "Error: ECC must be L, M, Q or H"
```

Comment:
It's fine but you could get some interesting result if there's a finally clause as that will be executed still.